webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/OrderedScrambles.java

```diff
@@ -0,0 +1,245 @@
+                for (JsonElement activity : room.getAsJsonObject().getAsJsonArray("activities")) {
+                    String activityCode = removeQuotation(activity.getAsJsonObject().get("activityCode").toString());
+
+                    String[] activityList = activityCode.split("-");
+                    String eventId = activityList[0];
+
+                    if (!eventId.equals(wcifIgnorableKey)) {
+
+                        DateTime activityStartTime = new DateTime(removeQuotation(activity.getAsJsonObject().get("startTime").toString()), timezone);
+
+                        int activityDay = Days.daysBetween(competitionStartDate.withTimeAtStartOfDay(), activityStartTime.withTimeAtStartOfDay()).getDays()+1;
+                        
+                        if (!dayList.contains(activityDay)) {
+                            dayList.add(activityDay);
+                            scrambleRequestListByDay.add(new ArrayList<ScrambleRequest>());
+                        }
+
+                        int round = -1;
+                        int group = -1;
+                        int attempt = -1;
+                        
+                        // This part assumes every round, group and attempt is labeled with an integer from competitionJson
+                        for (String item : activityList) {
+                            if (item.charAt(0) == 'r') {
+                                round = Integer.parseInt(item.substring(1));
+                            } else if (item.charAt(0) == 'g') {
+                                group = Integer.parseInt(item.substring(1));
+                            } else if (item.charAt(0) == 'a') {
+                                attempt = Integer.parseInt(item.substring(1));
+                            }
+                        }
+
+                        ScrambleRequest[] scrambleRequests = GSON.fromJson(sheet, ScrambleRequest[].class);
+                        
+                        // first, we add all requests whose events equals what we need
+                        ArrayList<ScrambleRequest> scrambleRequestTemp = new ArrayList<ScrambleRequest>();
+                        for (ScrambleRequest item : scrambleRequests) {
+                            if ((item.event).equals(eventId)) {
+                                scrambleRequestTemp.add(item);
+                            }
+                        }
+
+                        // then, we start removing
+                        if (round > 0) {
+                            ArrayList<ScrambleRequest> temp = new ArrayList<ScrambleRequest>();
+                            for (ScrambleRequest scrambleRequest : scrambleRequestTemp) {
+                                if (scrambleRequest.round == round) {
+                                    temp.add(scrambleRequest);
+                                }
+                            }
+                            scrambleRequestTemp = new ArrayList<ScrambleRequest>(temp);
```

Comment:
There are enough variables with the name "temp" in them that I'm getting confused. Can you think of a better name for `scrambleRequestTemp`?

update: After reading all this code, I think the high level idea here is that you have to search over the existing scramble requests to find the ones that match this wcif activity? It might be cleaner to pull that out into its own function... something that takes the wcif activity and the list of all scramble requests and returns the scramble requests that match that activity?

Reply:
Done. New WCIFHandler class.